src/libraries/System.Threading.Tasks.Parallel/tests/ParallelForEachAsyncTests.cs

```diff
@@ -835,6 +835,24 @@ public async Task Exception_FromDispose_Async()
             Assert.True(t.IsFaulted);
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        public async Task Exception_FromDisposeAndCancellationCallback_Sync()
+        {
+            Task t = Parallel.ForEachAsync((IEnumerable<int>)new ThrowsExceptionFromDisposeAndCancellationCallback(), (item, cancellationToken) => default);
+            await Assert.ThrowsAsync<FormatException>(() => t);
+            Assert.True(t.IsFaulted);
+            Assert.DoesNotContain(t.Exception.InnerExceptions, e => e is TimeZoneNotFoundException);
```

Comment:
Is `TimeZoneNotFoundException` correct? I don't see it anywhere in the file. Shouldn't it be `InvalidTimeZoneException`?

Reply:
Yup.